birdie/src-tauri/src/repo/locks.rs

```diff
@@ -166,10 +179,54 @@ async fn internal_lock_handler(
 ) -> Result<Json<LockResponse>, CoreError> {
     let github_pat = state.app_config.read().ensure_github_pat()?;
 
+    // make a new vec of paths
+    // for each path in self.paths, if path is a directory, add all files in the directory recursively to the vec
+    // if path is a file, add the file to the vec
+    let git_client = state.git();
+    let repo_path = git_client
+        .repo_path
+        .to_str()
+        .expect("was the git client passed an invalid repo path?");
+    let mut paths = vec![];
+    for path in &request.paths {
+        let full_path = std::path::Path::new(repo_path).join(path);
+        if full_path.is_dir() {
+            for entry in WalkDir::new(full_path) {
+                let entry = entry?;
+                if entry.file_type().is_file() {
+                    // remove repo dir from path
+                    let entry = entry.path().strip_prefix(repo_path)?;
+                    paths.push(entry.to_string_lossy().to_string().replace("\\", "/"));
+                }
```

Comment:
`WalkDir` recursively walks the file tree and lists everything, files and directories included. so for example if there were a folder structure like:
```
pizza/
   foo.txt
   bar.txt
 ```
 
 WalkDir would list `pizza`, `pizza/foo.txt`, and `bar.txt`. So in this case, we can simply skip anything that's a directory, because we know we're getting all the file nodes underneath anyway.